cameo/strain_design/heuristic/evolutionary/optimization.py

```diff
@@ -205,7 +229,21 @@ def run(self, evaluator=None, generator=None, view=config.default_view, maximize
 
         for observer in self.observers:
             observer.reset()
+
         t = time.time()
+
+        if max_time is not None:
+            terminator = self.heuristic_method.terminator
```

Comment:
What if `self.heuristic_method.terminator` is already `inspyred.ec.terminators.time_termination`?